mlflow/store/artifact/databricks_models_artifact_repo.py

```diff
@@ -119,14 +124,58 @@ def _extract_headers_from_signed_url(self, headers):
         filtered_headers = filter(lambda h: "name" in h and "value" in h, headers)
         return {header.get("name"): header.get("value") for header in filtered_headers}
 
+    def _parallelized_download_from_cloud(
+        self, signed_uri, headers, file_size, dst_local_file_path, dst_run_relative_artifact_path
+    ):
+        try:
+            failed_downloads = parallelized_download_file_using_http_uri(
+                http_uri=signed_uri,
+                download_path=dst_local_file_path,
+                file_size=file_size,
+                # URI type is not known in this context
+                uri_type=None,
+                chunk_size=_DOWNLOAD_CHUNK_SIZE,
+                headers=headers,
+            )
+            if failed_downloads:
+                new_signed_uri, new_headers = self._get_signed_download_uri(
+                    dst_run_relative_artifact_path
+                )
+            for i, excep in failed_downloads.items():
+                if excep.response.status_code not in (401, 403):
```

Comment:
Can we throw if `failed_downloads` contains an error whose status code is not 401 or 403 to avoid useless download?

```python
if any(e not in (401, 403)  for e in failed_downloads.values()):
     raise ...

for i, excep in failed_downloads.items():
    download_chunk(....)
```

Reply:
+1. We should fail early here before retrying downloading failed chunks.

Reply:
Good point!